tests/integration/synapseclient/test_evaluations.py

```diff
@@ -110,7 +110,7 @@ async def test_evaluations(syn: Synapse, project: Project):
         submissions = syn.getSubmissions(ev, limit=num_of_submissions - 1)
         for submission in submissions:
             assert re.match("Submission \\d+", submission["name"])
-            status = syn.getSubmissionStatus(submission)
+            status = syn.getSubmissionStatus(submission.get("id"))
```

Comment:
This used to work because `id_of` has some functionality to handle mapping objects like dictionaries. Because we are now validating for non-decimal submission IDs, we're no longer using `id_of`, meaning `getSubmissionStatus` and `getSubmission` no longer support mapping.

Supporting mapping was not part of their intended functionality based on the docstring for `getSubmission`:

```
"""
Gets a [synapseclient.evaluation.Submission][] object by its id.

Arguments:
    id: The id of the submission to retrieve
```

so this is not a feature change.

Reply:
This is a feature change, because people are no longer able to pass in the `Submission` object

Reply:
So I realize there is some confusion, but the docstring was basically wrong, it didn't just take id.  Even if that may have been the intended case for the function, the reality is we did support these the `Submission` object.